docs/mdx-plugins/remark-link-rewrite.test.js

```diff
@@ -0,0 +1,103 @@
+import path from 'path';
+
+import linkRewrite from './remark-link-rewrite';
+
+// helper functions to construct remark nodes
+const makeFile = filePath => ({
+  cwd: '/absolue/path/to/docs',
+  history: [path.join('/absolue/path/to/docs/pages', filePath)],
+});
+const rewrite = (file, url) => {
+  const link = { type: 'link', url };
+  const tree = { type: 'root', children: [link] };
+  linkRewrite()(tree, file);
+  return link.url;
+};
+
+describe('only rewrites internal links', () => {
+  const file = makeFile('index.md');
+
+  it('skips https://external.link', () => {
+    expect(rewrite(file, 'https://external.link')).toBe('https://external.link');
+  });
+
+  it('skips http://external.link?some=query', () => {
+    expect(rewrite(file, 'http://external.link?some=query')).toBe(
+      'http://external.link?some=query'
+    );
+  });
+});
+
+describe('from index page', () => {
+  const file = makeFile('index.md');
+
+  it('resolves guides to guides/', () => {
+    expect(rewrite(file, 'guides')).toBe('guides');
+  });
```

Comment:
The name of the test is confusing because of that trailing slash, which then doesn't appear in assertion.